### ISSUE TITLE
Integrate Font Awesome 5 (with 4.7 option)

### ISSUE THREAD

**Issue by 0dp** [understrap/understrap#586]
> This post is subject to updates

Integrating Font Awesome into UnderStrap can be done in the following ways: 

## Manually, documented in the docs
Method of manual integration has been discussed and proved here #547 
This method should be documented regardless, in case a theme developer wants to change from FA4.7 to FA5 free or FA5 pro, or a developer wants to change from FA5 free to FA5 pro.

## Automated, via npm install script
NPM packages (package.json) supports `preinstall` scripts which will allow us to modify the package.json file before it is processed. ie. changing dependencies. Theoretically we should be able to setup the entire project according to #547 
Eventually, it could possibly also allow developers installing 3rd party components per project basis.
My only concern about this approach is that it does not seem to be a common way of configuring npm packages. available examples?

## Automated, via scaffolding systems like Yeoman (others?)

Yeoman would add an extra layer to the development process, but unlike npm preinstall scripts, it's made for scaffolding. 

```
What's Yeoman?

Yeoman helps you to kickstart new projects, prescribing best practices and tools to help you stay productive.

To do so, we provide a generator ecosystem. A generator is basically a plugin that can be run with the `yo` command to scaffold complete projects or useful parts.

Through our official Generators, we promote the "Yeoman workflow". This workflow is a robust and opinionated client-side stack, comprising tools and frameworks that can help developers quickly build beautiful web applications. We take care of providing everything needed to get started without any of the normal headaches associated with a manual setup.

With a modular architecture that can scale out of the box, we leverage the success and lessons learned from several open-source communities to ensure the stack developers use is as intelligent as possible.

As firm believers in good documentation and well thought out build processes, Yeoman includes support for linting, testing, minification and much more, so developers can focus on solutions rather than worrying about the little things.
```

There is already a generator available for underscores found [here](https://github.com/mnyorba/generator-npm-wp-s-theme) which could be used as template.

**Comment by holger1411** [comment]
Puh, lot of opinions and ideas here: https://github.com/understrap/understrap/issues/547
I created some basic rules for UnderStrap here:
https://github.com/orgs/understrap/teams/understrap-playbook/discussions/1

That just a few things which are important to me. Everyone is free to comment this and I will add more rules or change/shape the current four (as long as we can define a small set of up to 7 main rules).

If I check these (current rules) we should:
- Remove any FA integration from .php files. Bootstrap uses another way to render small icons (like in the dropdown component or pagination):
http://getbootstrap.com/docs/4.0/components/dropdowns/

- Keep the current FA.47 integration in the dev stack (package.json->gulpfile->scss->css) but make it more maintainable (<- more comments/description + `preinstall` )

**Comment by 0dp** [comment]
@holger1411 I really like the less is more philosophy and agree that it's not all about adding features, but refining the core of the project.

Therefor i concur.

- All FA integration should be removed from theme files and revert to the BS way. That said, it would not hurt if we had a small guide on how to re implement the chevron into the walker.

- FA is a feature of the theme and should stay in the dev stack. I hereby propose that we use the preinstall script to simply prompt whether a developer wants to have FA included at all or not (current version). I even have a working demo for that task.

- The documentation should include how to manually add and remove different versions of FA to begin with. We can add different icon sets along the way ie. the ones BS recommends.

For automated install of FAx-xxx at a later time, I/we really need to do more research and discuss our options. 

I will wait for you comments on the revised bullet points and update OP.

My brain hurt so I will leave this for now 😸

**Comment by axlright** [comment]
I agree. . . I don't know how to type with emojis, but if I did, I'd insert something clever here >><<

**Comment by 0dp** [comment]
@axlright  🍆

**Comment by raisonon** [comment]
TL;DR: I agree with removing and if any integration is done wait a while for FA to get it's act together.

-- 
Long: I've just upgraded to v.5 on my local install. The whole process is really messy with FA at the moment. 

It surprised me. NPM repo still on 4.7 so had to pull from https://www.npmjs.com/package/@fortawesome/fontawesome-free-webfonts

The FA docs are no good and took longer than expected.

Just commenting that FA probably needs a few months for them to standardise, get npm sorted etc..  Any integration with 5 would probably require a re-jig in a few months, so probably best to hold off and ideally remove as @holger1411 suggests.

I've also had some clients use Google Material Icons rather than FA which come with SVGs. SVGs are favoured to icons (can have multiple colours, can transform and better for screen readers).  So perhaps ditching font-icons altogether is a good route?

Hope that context helps.

**Comment by 0dp** [comment]
@fortawesome will/is the new repo.
Docs are messy yes, but we deciphered it and came up with a step by step in another thread.

Coming up with steps to integrate other iconsets into understrap is very welcomed in the documentation :)

**Comment by 0dp** [comment]
@holger1411 
We are working on the documentation for including FA5 into the project. Likewise I will write a small guide on FA4, mostly on how to remove it.

So as of now we are going with Option #1 - Manually, documented in the docs

**Comment by Thomas-A-Reinert** [comment]
Will be included with the upcoming documentation. Closing THIS issue for housekeeping as it seems that common consent has been made? Feel free to reopen.

**Comment by PeterSafwat** [comment]
Any News About upgrading to FA5 ?

**Comment by xicubed** [comment]
I just left everything default and inserted this into my head which gave me access to all the free 5.4.2 icons.

`<link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.4.2/css/all.css" integrity="sha384-/rXc/GQVaYpyDdyxK+ecHPVYJSN9bmVFBvjA/9eOB+pb3F2w2N6fc5qB9Ew5yIns" crossorigin="anonymous">`

via https://fontawesome.com/how-to-use/on-the-web/setup/getting-started?using=web-fonts-with-css
